pkg/gits/provider_fake.go

```diff
@@ -723,24 +768,65 @@ func (r *FakeRepository) Name() string {
 }
 
 // NewFakeRepository creates a new fake repository
-func NewFakeRepository(owner string, repoName string) *FakeRepository {
-	return &FakeRepository{
+func NewFakeRepository(owner string, repoName string, addFiles func(dir string) error, gitter Gitter) (*FakeRepository, error) {
```

Comment:
Would it not be better to have a new function instead here? Something like `NewLocalRepository`? IMO this repo is not a fake per se anymore and by adding _addFiles_ and _gitter_ here you need to adjust a fair bit of existing tests passing effectively _nil_s.

Reply:
That adjustment is now done anyway - that's what was needed to make the tests pass.